1 test fails: AssertionError: assert ['second', 'first'] == ['first', 'second']

```
========================================================================================== ERRORS ===========================================================================================
______________________________________________________________________ ERROR at teardown of TestRunAllTests.test_first ______________________________________________________________________

cls = <class 'nibabel.tests.test_api_validators.TestRunAllTests'>

    @classmethod
    def teardown_class(cls):
        # Check that both validate_xxx tests got run
>       assert cls.run_tests == ['first', 'second']
E       AssertionError: assert ['second', 'first'] == ['first', 'second']
E         
E         At index 0 diff: 'second' != 'first'
E         
E         Full diff:
E           [
E         +     'second',
E               'first',
E         -     'second',
E           ]

nibabel/tests/test_api_validators.py:116: AssertionError
========================================================================================= FAILURES ==========================================================================================
__________________________________________________________________________________ TestEcatImage.test_file __________________________________________________________________________________

self = <nibabel.tests.test_ecat.TestEcatImage testMethod=test_file>

    def test_file(self):
>       assert Path(self.img.file_map['header'].filename) == Path(self.example_file)
E       AssertionError: assert PosixPath('tinypet_tmp.v') == PosixPath('/usr/ports/science/py-nibabel/work-py311/nibabel-5.3.0/nibabel/tests/data/tinypet.v')
E        +  where PosixPath('tinypet_tmp.v') = Path('tinypet_tmp.v')
E        +    where 'tinypet_tmp.v' = <nibabel.fileholders.FileHolder object at 0x3d17265d0650>.filename
E        +  and   PosixPath('/usr/ports/science/py-nibabel/work-py311/nibabel-5.3.0/nibabel/tests/data/tinypet.v') = Path('/usr/ports/science/py-nibabel/work-py311/nibabel-5.3.0/nibabel/tests/data/tinypet.v')
E        +    where '/usr/ports/science/py-nibabel/work-py311/nibabel-5.3.0/nibabel/tests/data/tinypet.v' = <nibabel.tests.test_ecat.TestEcatImage testMethod=test_file>.example_file

nibabel/tests/test_ecat.py:186: AssertionError
```

Version: 5.3.0
Python-3.11
FreeBSD 14.1

--- a/nibabel/tests/test_api_validators.py
+++ b/nibabel/tests/test_api_validators.py
@@ -99,18 +99,18 @@ class TestRunAllTests(ValidateAPI):
     We check this in the module teardown function
     """
 
-    run_tests = []
+    run_tests = {}
 
     def obj_params(self):
         yield 1, 2
 
     def validate_first(self, obj, param):
-        self.run_tests.append('first')
+        self.run_tests.add('first')
 
     def validate_second(self, obj, param):
-        self.run_tests.append('second')
+        self.run_tests.add('second')
 
     @classmethod
     def teardown_class(cls):
         # Check that both validate_xxx tests got run
-        assert cls.run_tests == ['first', 'second']
+        assert cls.run_tests == {'first', 'second'}

